File: windows/DMShot.Tests/ScreenshotFilenameTests.cs
using System;
using System.Collections.Generic;
using DMShot.Editor;
using Xunit;

public class ScreenshotFilenameTests
{
    [Fact]
    public void Base_FormatsDDMMYYYY_HH_MM()
    {
        var name = ScreenshotFilename.Base(new DateTime(2026, 6, 18, 14, 30, 0));
        Assert.Equal("DM_Screenshot_18062026_14_30", name);
    }

    [Fact]
    public void Base_ZeroPads()
    {
        var name = ScreenshotFilename.Base(new DateTime(2026, 1, 3, 9, 5, 0));
        Assert.Equal("DM_Screenshot_03012026_09_05", name);
    }

    [Fact]
    public void Unique_ReturnsBaseWhenFree()
    {
        var name = ScreenshotFilename.Unique("DM_Screenshot_18062026_14_30", _ => false);
        Assert.Equal("DM_Screenshot_18062026_14_30.png", name);
    }

    [Fact]
    public void Unique_AppendsSuffixOnCollision()
    {
        var taken = new HashSet<string>
        {
            "DM_Screenshot_18062026_14_30.png",
            "DM_Screenshot_18062026_14_30_1.png",
        };
        var name = ScreenshotFilename.Unique("DM_Screenshot_18062026_14_30", taken.Contains);
        Assert.Equal("DM_Screenshot_18062026_14_30_2.png", name);
    }
}

File: mac/Tests/DMShotTests/ScreenshotFilenameTests.swift
import XCTest
@testable import DMShot

final class ScreenshotFilenameTests: XCTestCase {
    private func date(_ d: Int, _ m: Int, _ y: Int, _ h: Int, _ min: Int) -> Date {
        var c = DateComponents()
        (c.day, c.month, c.year, c.hour, c.minute) = (d, m, y, h, min)
        return Calendar(identifier: .gregorian).date(from: c)!
    }

    func testBaseNameFormatDDMMYYYY_HH_MM() {
        let base = ScreenshotFilename.base(for: date(18, 6, 2026, 14, 30))
        XCTAssertEqual(base, "DM_Screenshot_18062026_14_30")
    }

    func testBaseNameZeroPads() {
        let base = ScreenshotFilename.base(for: date(3, 1, 2026, 9, 5))
        XCTAssertEqual(base, "DM_Screenshot_03012026_09_05")
    }

    func testUniqueReturnsBaseWhenFree() {
        let name = ScreenshotFilename.unique(base: "DM_Screenshot_18062026_14_30") { _ in false }
        XCTAssertEqual(name, "DM_Screenshot_18062026_14_30.png")
    }

    func testUniqueAppendsSuffixOnCollision() {
        let taken: Set<String> = ["DM_Screenshot_18062026_14_30.png",
                                  "DM_Screenshot_18062026_14_30_1.png"]
        let name = ScreenshotFilename.unique(base: "DM_Screenshot_18062026_14_30") {
            taken.contains($0)
        }
        XCTAssertEqual(name, "DM_Screenshot_18062026_14_30_2.png")
    }
}
